src/types/type_param.rs

```diff
@@ -56,6 +56,19 @@ pub enum TypeArg {
     CustomValue(serde_yaml::Value),
 }
 
+impl TypeArg {
+    /// Report [`TypeArg`] if param is a type
+    pub fn tag(&self) -> Option<TypeTag> {
+        match self {
+            TypeArg::Type(s) => Some(s.tag()),
+            TypeArg::ClassicType(c) => Some(c.tag()),
+            // assume list is well formed - all elements of same type
```

Comment:
I *think* this is a "kinding error" (Conor's term), i.e. this is operating at the wrong level of types vs values (vs kinds etc.)

E.g., if the TypeParam was `List<ClassicType>`, then two valid instantiations might be
1. `[ClassicType::Float, ClassicType::Hashable(HashableType::Int), .....]`
2. `[ClassicType::Hashable(HashableType::String), ClassicType::Hashable(HashableType::Int)].

So the "right" thing to do is probably to take the `union` of all the elements, which would declare Classic for the first example, but hashable for the second.

However, I'd also be happy to drop this case and just return None

Reply:
yes agreed - I've gone with None since I'm not convinced treating List[TypeArg] implicitly as a Type is the right thing to do.